### PR TITLE
Re-enable elementAccessorDump in swift

### PR DESCRIPTION
**Pull request by ianfixes**
* Quickly converts `XCUIApplication().debugDescription` to a copy-pastable list of element strings
* provides conversion from these elements to their actual underlying `XCUIElement`

```
app
app.navigationBars["Home"]
app.navigationBars["Home"].images.elementAtIndex(0)
app.navigationBars["Home"].searchFields["Search Site"]
```

### CODE REVIEW

File: Pod/Illuminator/IlluminatorElement.swift [hunk 1]
```diff
@@ -0,0 +1,425 @@
...
+        }
+    }
+    
+    
+    static func elementTypeFromString(_ description: String) -> XCUIElementType {
+        guard let val = theXCUIElementTypeOfDebugString[description] else { return .Other }
+        return val
+    }
+    
+    static func stringFromElementType(_ theType: XCUIElementType) -> String {
```
Comment (kviksilver) [inline Pod/Illuminator/IlluminatorElement.swift]: Would make these live on `XCUIElementType` together with `elementTypeFromString` `elementStringFromElementType` `theXCUIElementTypeOfDebugString` `debugStringOfXCUIElementType`

File: Pod/Illuminator/IlluminatorElement.swift [hunk 2]
```diff
@@ -0,0 +1,425 @@
...
+        } catch AbortIlluminatorTree.parseError(let badLine) {
+            print("Caught a parse error of \(badLine)")
+        } catch {
+            print("Caught an error that we didn't throw... somehow")
+        }
+        return nil
+    }
+    
+    // return a tree of IlluminatorElements from a debugDescription
+    static func fromDebugDescription(_ content: String) -> IlluminatorElement? {
```
Comment (kviksilver) [inline Pod/Illuminator/IlluminatorElement.swift]: initializer?

**Comment by ianfixes** [comment]
@kviksilver I should probably update IlluminatorTestProgress (https://github.com/ifreecarve/Illuminator/blob/2016-12-19_illuminator_element/Pod/Illuminator/IlluminatorTestProgress.swift#L101)

So that the user can specify their own on-pass/on-fail behavior.  Any idea for a pattern on that?

File: Pod/Illuminator/IlluminatorElement.swift [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+        return elem
+    }
+    
+    // return a tree of IlluminatorElements from the relevant section of a debugDescription
+    // or nil if there is a parse error
+    private static func parseTree(_ content: String) -> IlluminatorElement? {
+        let lines = content.componentsSeparatedByString("\n")
+        guard lines.count > 0 else { return nil }
+        
+        print("regex all elements")
```
Comment (kviksilver) [inline Pod/Illuminator/IlluminatorElement.swift]: Do we need this print statement? and one on L: 256

**Review by kviksilver — APPROVED** [review]

File: Pod/Illuminator/IlluminatorElement.swift (hunk 1 above)
Comment (ifreecarve) [inline Pod/Illuminator/IlluminatorElement.swift]: I don't think this is possible since `XCUIElementType` appears to be an enum inside `XCUIElement`

**Pull request merged**